extensions/sequence.md

```diff
@@ -0,0 +1,51 @@
+# Sequence
+
+This extension defines two attributes that can be included within a CloudEvent
+to describe the position of the event relative to the other events from the
+same event source.
+
+The `sequence` attribute represents the value of this event's
+order in the stream of events.  The exact value and meaning of this
+attribute is defined by the `sequenceType` attribute. 
+If the `sequenceType` is missing, or not defined in this specification,
+event consumers will need to have some out-of-band communication with the
+event producer to understand how to interpret the value of the attribute.
+
+## Attributes
+
+### sequence
+* Type: `String`
+* Description: Value expressing the relative order of the event. This enables
+  interpretation of data supercedence.
```

Comment:
I wonder if this would be better done by checking the EventID instead?  If not, have any suggested text?
Do we need to worry about a single event being part of more than one sequence?

Reply:
> I wonder if this would be better done by checking the EventID instead?

That's a great question.  As per [the eventID definition](https://github.com/cloudevents/spec/blob/master/spec.md#eventid) there is no monotinicity requirement.  Without a value constraint, one would have to keep track of every `eventID` that has been processed (or at the very least a significant scope of them) to ensure no duplicate processing as opposed to checking the relationship of the currently being processed `eventID` to the last successful checkpoint `eventID`.  I'm not sure that the assumptions I like to make would be appropriate to push onto all users fo the spec so I haven't advocated for that there but would have them adopted in my more-perfect world.

> Do we need to worry about a single event being part of more than one sequence?

I'm taking an event as the tangible message sent by a unique producer (as opposed to the occurrence in reality being read by some sensor).  As I've read this change, it seemed that there was a one-to-one mapping between unique producers of events with this attribute and the series of numbers representing the singular sequence that they produce.  That's a long way to say I don't expect a unique producer to produce multiple sequences OR for multiple unique producers to produce the "same" event (though distinct events could be causally interlinked and contain the exact same content [sans producer id] by the shared reality producing them - e.g. redundant emitters of metrics coming from a single sensor).  I could definitely be reading incorrectly.